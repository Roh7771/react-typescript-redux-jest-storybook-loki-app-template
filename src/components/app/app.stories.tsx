import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { action } from '@storybook/addon-actions';
import { Provider } from 'react-redux';
import App, { AppProps } from './app';
import { FullStateType } from '../../types/general-types';

const store: any = {
  getState: (): FullStateType => ({
    data: {
      title: 'Hello World!!!',
    },
  }),
  subscribe: () => () => null,
  dispatch: action('dispatch'),
};

export default {
  components: App,
  decorators: [story => <Provider store={store}>{story()}</Provider>],
  title: 'App',
} as Meta;

const Template: Story<AppProps> = args => <App {...args} />;

export const Default = Template.bind({});
