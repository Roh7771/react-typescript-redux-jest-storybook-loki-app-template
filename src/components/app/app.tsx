import React from 'react';
import { connect } from 'react-redux';
import { ThunkDispatch } from 'redux-thunk';
import { DataActionCreators } from '../../reducer/data/data';
import getTitle from '../../reducer/data/selectors';
import { DataActionTypes } from '../../types/redux/data/types';
import { FullStateType, MockAxiosInstance } from '../../types/general-types';

export type AppProps = {
  title: string;

  onTitleChange: (title: string) => void;
  onTitleReset: () => void;
};

export const App: React.FunctionComponent<AppProps> = ({
  title,
  onTitleChange,
  onTitleReset,
}: AppProps) => (
  <div>
    <h1 className="title">{title}</h1>
    <button type="button" onClick={() => onTitleChange('Another Title')}>
      Change title text to Another Title
    </button>
    <button type="button" onClick={() => onTitleReset()}>
      Reset title
    </button>
  </div>
);

const mapStateToProps = (state: FullStateType) => ({
  title: getTitle(state),
});

const mapDispatchToProps = (
  dispatch: ThunkDispatch<FullStateType, MockAxiosInstance, DataActionTypes>,
) => ({
  onTitleChange: (title: string) => {
    dispatch(DataActionCreators.setTitle(title));
  },
  onTitleReset: () => {
    dispatch(DataActionCreators.resetTitle());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
