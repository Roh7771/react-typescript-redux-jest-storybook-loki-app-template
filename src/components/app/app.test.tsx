import React from 'react';
import renderer from 'react-test-renderer';
import { App, AppProps } from './app';

const mockProps: AppProps = {
  title: 'Some Title',
  onTitleChange: jest.fn(),
  onTitleReset: jest.fn(),
};

test('App should render correct', () => {
  const tree = renderer.create(<App {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
