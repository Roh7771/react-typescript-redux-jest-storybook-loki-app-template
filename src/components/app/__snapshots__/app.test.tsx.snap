// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correct 1`] = `
<div>
  <h1
    className="title"
  >
    Some Title
  </h1>
  <button
    onClick={[Function]}
    type="button"
  >
    Change title text to Another Title
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Reset title
  </button>
</div>
`;
